apps/api/v2/src/modules/organizations/controllers/schedules/organizations-schedules.controller.ts

```diff
@@ -0,0 +1,134 @@
+import { SchedulesService_2024_06_11 } from "@/ee/schedules/schedules_2024_06_11/services/schedules.service";
+import { API_VERSIONS_VALUES } from "@/lib/api-versions";
+import { Roles } from "@/modules/auth/decorators/roles/roles.decorator";
+import { ApiAuthGuard } from "@/modules/auth/guards/api-auth/api-auth.guard";
+import { IsOrgGuard } from "@/modules/auth/guards/organizations/is-org.guard";
+import { RolesGuard } from "@/modules/auth/guards/roles/roles.guard";
+import { IsUserInOrg } from "@/modules/auth/guards/users/is-user-in-org.guard";
+import { OrganizationsSchedulesService } from "@/modules/organizations/services/organizations-schedules.service";
+import {
+  Controller,
+  UseGuards,
+  Get,
+  Post,
+  Param,
+  ParseIntPipe,
+  Body,
+  Patch,
+  Delete,
+  HttpCode,
+  HttpStatus,
+} from "@nestjs/common";
+import { ApiTags as DocsTags } from "@nestjs/swagger";
+
+import { SUCCESS_STATUS } from "@calcom/platform-constants";
+import {
+  CreateScheduleInput_2024_06_11,
+  CreateScheduleOutput_2024_06_11,
+  DeleteScheduleOutput_2024_06_11,
+  GetScheduleOutput_2024_06_11,
+  GetSchedulesOutput_2024_06_11,
+  UpdateScheduleInput_2024_06_11,
+  UpdateScheduleOutput_2024_06_11,
+} from "@calcom/platform-types";
+
+@Controller({
+  path: "/v2/organizations/:orgId",
+  version: API_VERSIONS_VALUES,
+})
+@UseGuards(ApiAuthGuard, IsOrgGuard, RolesGuard)
+@DocsTags("Organizations Schedules")
+export class OrganizationsSchedulesController {
+  constructor(
+    private schedulesService: SchedulesService_2024_06_11,
+    private organizationScheduleService: OrganizationsSchedulesService
+  ) {}
+
+  @Roles("ORG_ADMIN")
+  @Get("/schedules")
+  async getOrganizationSchedules(
+    @Param("orgId", ParseIntPipe) orgId: number
+  ): Promise<GetSchedulesOutput_2024_06_11> {
+    const schedules = await this.organizationScheduleService.getOrganizationSchedules(orgId);
+
+    return {
+      status: SUCCESS_STATUS,
+      data: schedules,
+    };
+  }
```

Comment:
we might need a pagination system here, that impact the amount of userIds selected

Reply:
see here https://github.com/calcom/cal.com/pull/15586/files#diff-f8bfbca9027ce1b0a8ef248a14b6ed60f92f3f5fe7d58e4c8ca621d5fcc4416f

Reply:
Implemented in [feat: add pagination to getOrganizationSchedules](https://github.com/calcom/cal.com/pull/15584/commits/c0568286a19deb494a942d3b0c0d1f2b2693386c) which resulted in adding new [organizations-schedules.service.ts](https://github.com/calcom/cal.com/pull/15584/files#diff-e1f408ee6b5d9b3fec2e2d5d557c0c89b94b5902a3230d90a8002e61a9c2aa31) and [organizations-schedules.repository.ts](https://github.com/calcom/cal.com/pull/15584/files#diff-8a139b57d03971931ab1fa52d103c0d7e6520b66f7ab1e9b23e097b51e7a061e).